cmake/Modules/Packages.cmake

```diff
@@ -450,17 +450,25 @@ endif()
 
 #set(DEV_WARNINGS ${CMAKE_SUPPRESS_DEVELOPER_WARNINGS})
 
-checkout_git_submodule(RECURSIVE
-    RELATIVE_PATH external/cereal
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    REPO_URL https://github.com/jrmadsen/cereal.git
-    REPO_BRANCH timemory)
+if (TIMEMORY_BUILD_CEREAL)
+    checkout_git_submodule(RECURSIVE
+        RELATIVE_PATH external/cereal
+        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        REPO_URL https://github.com/jrmadsen/cereal.git
+        REPO_BRANCH timemory)
 
-# add cereal
-add_subdirectory(${PROJECT_SOURCE_DIR}/external/cereal)
+    # add cereal
+    add_subdirectory(${PROJECT_SOURCE_DIR}/external/cereal)
 
-target_include_directories(timemory-cereal SYSTEM INTERFACE
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/external/cereal/include>)
+    target_include_directories(timemory-cereal SYSTEM INTERFACE
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/external/cereal/include>)
+else()
+    find_package(cereal CONFIG REQUIRED)
+    get_target_property(TIMEMORY_CEREAL_INCLUDE cereal INTERFACE_INCLUDE_DIRECTORIES)
+
+    message(STATUS "including TIMEMORY_CEREAL_INCLUDE=${TIMEMORY_CEREAL_INCLUDE}")
+    target_include_directories(timemory-compile-options INTERFACE ${TIMEMORY_CEREAL_INCLUDE})
```

Comment:
This should be:

```cmake
target_include_directories(timemory-cereal INTERFACE ${TIMEMORY_CEREAL_INCLUDE})
```

Each TPL has it's own interface library and later there is `target_link_libraries(timemory-headers INTERFACE timemory-cereal)`. Thus you can link to `timemory::timemory-headers` in cmake and you always get `timemory::timemory-cereal`. The `timemory-compile-options` interface library is just for compiler flags internal to the library and won't necessarily be included in projects using timemory downstream.

Reply:
I am just going to accept this since it will be removed soon in #101